src/control/events/generic.go

```diff
@@ -47,3 +49,15 @@ func StrInfoToProto(si *StrInfo) (*sharedpb.RASEvent_StrInfo, error) {
 
 	return pbInfo, nil
 }
+
+// NewGenericEvent returns a generic RAS event with supplied ID.
+func NewGenericEvent(id RASID, sev RASSeverityID, msg, info string) *RASEvent {
```

Comment:
Is this something that would be used by production code? i.e. Will we ever want to raise a truly generic RAS event with a nil rank, etc? If not, I'd prefer to keep this unexported.